keps/sig-network/2634-kubeproxy-staging/README.md

```diff
@@ -0,0 +1,187 @@
+follow their own dataplane implementation (IPTables, IPVS, eBPF, open flow programming), 
+every implementation needs to follow the same logic: watching Services, Endpoints and 
+most recently EndpointSlices.
+
+Everytime we change the logic from Services, each of those implementations need to make 
+changes into their code base to be compliant with the new features. Some of them vendor
+the whole kubernetes/kubernetes repo, while others copy all the code base to their own
+repository.
+
+Moving pkg/proxy logics to its own repo will allow a better decoupling of the code, 
+the evolution of proxy features without being tighten to kubernetes main code and also 
+to future kube-proxy implementations to import this logic without needing to import the 
+whole Kubernetes code.
+
+This was also motivated by [kubernetes #92369](https://github.com/kubernetes/kubernetes/issues/92369)
+
+### Goals
+
+* Move kube-proxy repo to `kubernetes/kube-proxy` repo using staging.
+* Decouple kube-proxy from internal kubernetes dependencies.
+
+### Non-Goals
+* Implementing interfaces for kube-proxy new implementations (like virtual-kubelet, 
+but for proxies). This can be addressed on a later KEP
+
+## Proposal
+
+kube-proxy already have its own repo. The proposal is to:
+
+### Move the internal network dependencies that kube-proxy already has
+kube-proxy have some internal dependencies, mostly notable in `kubernetes/pkg/utils`. We
+can nominate some of them being `iptables`, `ipvs` and `conntrack` packages, but there are
+others as well.
+
+We need to move these dependencies to some repo that does not force kube-proxy to also
+vendor the whole kubernetes repo.
+
+<<[UNRESOLVED repository for network utils ]>>
+Although the majority of utils used by kube-proxy can be moved to component-helpers,
+some still vendors external dependencies (like `moby/ipvs`).
+If we move every pkg/utils used by kube-proxy to component helpers, another project
+already vendoring it (like kubectl) might vendor moby/ipvs because of transitive 
+dependencies.
+We need to define if network utils should be moved to its own repo (suggesting kubernetes/net-utils)
+or if we can accept that other projects vendoring `component-helpers` are going to vendor the 
+dependencies of kube-proxy as well
+<<[/UNRESOLVED]>>
+
+### Move other dependencies kube-proxy already has inside k/k
+kube-proxy vendors some packages that are used by other parts of kubernetes codebase. 
+From now, we can point:
+* pkg/util/async (async bounder) which is used by pkg/controlplane
+* pkg/util/sysctl which is used by pkg/kubelet and kubemark
+
+### Move feature gate package dependency
+Because kube-proxy relies on defined Feature gates, it vendors `pkg/features`.
+
+This needs to be copied/moved the same way controller manager already does, moving this to
+`kubernetes/kube-proxy/pkg/features` and changing on the code, and putting all the feature
+gates referenced by kube-proxy there
+
+### Move pkg/proxy to staging
```

Comment:
Should we also do cmd/kube-proxy?

Reply:
should we? not sure, I think that right now cmd/kube-proxy is in a good place where it is (following the same approach of kubelet/scheduler/etc) that are there.

But you are here on a really longer time that I am, so I may be missing something really important!!

Reply:
IMO the split between cmd/kube-proxy and pkg/proxy is pretty much completely artificial.  Nothing uses pkg/proxy except cmd/proxy (as far as I caan tell).  Repeat for kubelet, controllers, etc and their respective binaries.  It just makes the code harder to find.

That said, this is FAR from the most important thing going on right now, so I don't know if it makes sense to do a bunch of busywork?